src/relax/op/tensor/binary.cc

```diff
@@ -83,12 +92,17 @@ StructInfo InferStructInfoBroadcast(const Call& call, const BlockBuilder& ctx) {
     for (; i <= std::min(lhs_ndim, rhs_ndim); ++i) {
       const PrimExpr& dim0 = lhs_shape->values[lhs_ndim - i];
       const PrimExpr& dim1 = rhs_shape->values[rhs_ndim - i];
-      if (EqualConstInt(dim0, 1)) {
+      if (IsConstInt(dim0, 1)) {
         output_shape.push_back(dim1);
-      } else if (EqualConstInt(dim1, 1)) {
+      } else if (IsConstInt(dim1, 1)) {
         output_shape.push_back(dim0);
-      } else if (EqualCheck(dim0, dim1)) {
+      } else if (CanProveEqual(dim0, dim1)) {
```

Comment:
It is expensive to create analyzer on the fly for each dimension, consider add BlockBuilder.GetAnalyzer()

```c++
arith:: Analyzer* ana = ctx->GetAnalyzer();
ana->CanProveEqual(dim0, dim1)
```